### PR TITLE
Raise exception if the lockedgold value is zero.

### LINKED ISSUE #7758: [Bounty] Disallow zero value locked celo operations
### Expected Behavior

`GoldWithdrawn`, `GoldLocked`, `GoldRelocked`, `GoldUnlocked` should not emit when value is 0 (and corresponding operations should be prevented)

`AccountsSlashed`?

### Current Behavior

Locked Celo operations are used to trace internal celo transfers by indexers, which leads to a false positive when value is 0

Discovered by coinbase in rosetta operation
See fix: https://github.com/celo-org/rosetta/pull/160

### PR DESCRIPTION
**Pull request by deepakhb2**
### Description

GoldWithdrawn, GoldLocked, GoldRelocked, GoldUnlocked should not emit when value is 0 (and corresponding operations should be prevented)

### Other changes

NIL

### Tested

No new tests added but all existing unit tests pass after change.

### Related issues

- Fixes #7758 

### Backwards compatibility

Is backward compatible

### Documentation

Not required.

### CODE REVIEW

File: packages/protocol/test/governance/voting/lockedgold.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ contract('LockedGold', (accounts: string[]) => {
   beforeEach(async () => {
     mockGoldToken = await MockGoldToken.new()
     accountsInstance = await Accounts.new()
-    lockedGold = await LockedGold.new()
+    lockedGold = await LockedGold.new(true)
```
Comment (yorhodes) [inline packages/protocol/test/governance/voting/lockedgold.ts]: we should add tests for reverting on `lock`, `relock`, and `unlock` when `value == 0`

**Review by yorhodes — CHANGES_REQUESTED** [review]
1 request but otherwise great!

File: packages/protocol/test/governance/voting/lockedgold.ts (hunk 1 above)
Comment (deepakhb2) [reply packages/protocol/test/governance/voting/lockedgold.ts]: @yorhodes I have added the tests for the above scenario.

**Review by yorhodes — APPROVED** [review]
LGTM!

**Comment by deepakhb2** [comment]
@yorhodes Is this PR still required? I can fix the failing tests. Thank you.

**Pull request closed without merging**